packages/react-badge/src/components/CounterBadge/CounterBadge.types.ts

```diff
@@ -0,0 +1,82 @@
+import { BadgeProps, BadgeShape, BadgeAppearance, BadgeState } from '../Badge/index';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export type CounterBadgeColors = 'accent' | 'warning' | 'important' | 'severe' | 'informative';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export interface CounterBadgeProps extends Omit<BadgeProps, 'appearance' | 'shape'> {
+  /**
+   * A Badge can be circular or rounded
+   * @defaultvalue circular
+   */
+  shape?: Extract<BadgeShape, 'rounded' | 'circular'>;
+
+  /**
+   * A Badge can be filled, ghost
+   * @defaultvalue filled
+   */
+  appearance?: Extract<BadgeAppearance, 'filled' | 'ghost'>;
+
+  /**
+   * A Badge can have color variations
+   * @defaultvalue accent
+   */
+  color?: CounterBadgeColors;
+
+  /**
+   * Max number to be displayed
+   * @defaultvalue 99
+   */
+  overflowCount?: number;
+
+  /**
+   * Value diplayed by the Badge
+   * @defaultvalue 0
+   */
+  count?: number;
+
+  /**
+   * If the badge should be shown when count is 0
+   * @defaultvalue false
+   */
+  showZero?: boolean;
+
+  /**
+   * If a dot badge should be displayed
+   * @defaultvalue false
+   */
+  dot?: boolean;
+}
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export interface CounterBadgeState extends BadgeState {
```

Comment:
You should omit `appearance` and `shape` from BadgeState so you can use the narrowed types for those.

Also, I'm planning to add a helper type to make defining the State type easier. But for now you could write this instead:
```typescript
export interface CounterBadgeState extends Omit<BadgeState, 'appearance' | 'shape'> &
  Required<Pick<CounterBadgeProps, 'appearance' | 'shape' | 'overflowCount' | 'count' | 'showZero' | 'dot'>>;
```

Reply:
Not entirely sure about this tbh, aren't `appearance` and `shape` also `CounterBadgeProps` and `BadgeProps` ?

why can't the narrowing down happen on prop level, this seems to be getting complicated ?

Reply:
to clarify... I wonder if extending state here too is needed ?

Reply:
I still feel like `export interface CounterBadgeState extends CounterBadgeProps` is more clean?